jscomp/main/jsoo_main.ml

```diff
@@ -56,6 +56,22 @@ end
      *    type: "error" // or "warning" or "info"
      *  }
 *)
+let mk_js_error loc msg = 
+  let (file,line,startchar) = Location.get_pos_info loc.Location.loc_start in
+  let (file,endline,endchar) = Location.get_pos_info loc.Location.loc_end in
+  Js.Unsafe.(obj
+      [|
+        "js_error_msg",
+          inject @@ Js.string (Printf.sprintf "Line %d, %d:\n  %s"  line startchar msg);
+            "row"    , inject (line - 1);
+            "column" , inject startchar;
+            "endRow" , inject (endline - 1);
+            "endColumn" , inject endchar;
+            "text" , inject @@ Js.string msg;
+            "type" , inject @@ Js.string "error"
+      |]
+    )
+
 let () =  
   Bs_conditional_initial.setup_env ();
```

Comment:
This looks good to me in general.
Can we make a separate file instead of enriching such file, like reason_main.ml, the reason is that 
currently this is file is buildable, when reason linked in this file may not be easily buildable, we can have two files : one with vanilla ocaml which has less deps, the other has reason which is larger

Reply:
Not sure what you mean by "when reason is linked, this file may not be easily buildable"? Do you mean because of the reliability of the Reason code?

But yes, we can totally do a "jsoo_reason_main.ml" file exposing the ocaml + reason interface as well

Reply:
In general we don't compile packed files(to make compile time less for productivity) during dev but we compile jsoo_main.ml during dev. A separate file would help save some compilation time

Reply:
@bobzhang So I would create a file `jscomp/main/jsoo_refmt_main.ml`. How would I configure ninja, or whatever part is responsible, to build this file the way you want it? I am not even sure how this is supposed to save compilation time. It looks like it is as fast as without reason?

Reply:
@ryyppy just add a new file to main and run ./scripts/ninja.js config. 

I think the best course of action would be jsoo_refmt_main.ml reusing functions from jsoo_main.ml instead of duplicating everything.